server/src/main/java/org/elasticsearch/index/engine/InternalEngineFactory.java

```diff
@@ -24,4 +24,9 @@
     public Engine newReadWriteEngine(EngineConfig config) {
         return new InternalEngine(config);
     }
+
+    @Override
+    public Engine newNoopEngine(EngineConfig config) {
```

Comment:
Do we want to make this a plugin? I was thinking built-in but I guess we could move it to a module if we wanted to, I wasn't sure if it's a good idea to do engine implementations inside of modules

Reply:
Looking at this again, it appears that `EnginePlugin` is part of the CCR branch, are there plans to factor it out into a separate PR that goes into master?

Reply:
I think it would be great if we could make it a module, I am always happy to see isolation when it makes sense and here I think it can. Note that CCR has an engine implementation inside a module. 😇

We have talked exactly about the possibility of reusing the engine plugin work in the context of closed indices many months ago so we could bring that work out of the CCR branch and into 6.x/master.

Reply:
Okay, I've removed the changes to the `EngineFactory` and I can use the pluggable one for the next PR (since this one doesn't hook the engine in yet)